k2/csrc/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 add_library(properties properties.cc)
 target_include_directories(properties PUBLIC ${CMAKE_SOURCE_DIR})
 target_compile_features(properties PUBLIC cxx_std_11)
+target_link_libraries(properties PUBLIC fsa_algo)
```

Comment:
I am trying to add

```
target_link_libraries(fsa_algo PUBLIC properties)
```
since I want to use `IsConnected()` in `TopSort`;
but this results in circular dependencies:

```
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "properties" of type SHARED_LIBRARY
    depends on "fsa_algo" (weak)
  "fsa_algo" of type SHARED_LIBRARY
    depends on "properties" (weak)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
```

-----

Should we remove all `add_library` and write a single `add_library` like the following

```
add_library(fsa
  fsa_equivalent.cc
  fsa_util.cc
  properties.cc
  ....
)
```

Reply:
I'm OK with this (maybe except `fsa_render`?). Finally I guess we will have two `.so` files for common FSA operations, one for CPU and another for GPU?